From ec1a98a965b7a392e8d9686e7fb356a97aa4c165 Mon Sep 17 00:00:00 2001
From: Huang Shan <huang.shan@freescale.com>
Date: Thu, 17 Apr 2014 15:46:42 +0800
Subject: [PATCH 59/70] ENGR00303832 Ethernet:fix proxy bypass function not
 work issue

Fill bypass field in new ProxyProperties instead of null to support proxy
bypass function.

Signed-off-by: Huang Shan <huang.shan@freescale.com>
---
 Ethernet/com/fsl/ethernet/EthernetManager.java | 1 +
 Ethernet/res/layout/advanced_setting.xml       | 4 ++--
 Ethernet/res/values/strings.xml                | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Ethernet/com/fsl/ethernet/EthernetManager.java b/Ethernet/com/fsl/ethernet/EthernetManager.java
index 0cbc4d9..8ba9a43 100644
--- a/Ethernet/com/fsl/ethernet/EthernetManager.java
+++ b/Ethernet/com/fsl/ethernet/EthernetManager.java
@@ -364,6 +364,7 @@ public class EthernetManager {
 
     public void setProxy(){
         String exclusionList = null;
+        exclusionList=getSharedPreProxyExclusionList();
         if (getSharedPreProxyAddress() == null || getSharedPreProxyPort() == null)
             return;
         LinkProperties lp = mTracker.getLinkProperties();
diff --git a/Ethernet/res/layout/advanced_setting.xml b/Ethernet/res/layout/advanced_setting.xml
index 5258e29..b6379cd 100644
--- a/Ethernet/res/layout/advanced_setting.xml
+++ b/Ethernet/res/layout/advanced_setting.xml
@@ -86,8 +86,8 @@
                       android:layout_height="wrap_content"
                       android:textSize="18sp"
                       android:textAlignment="viewStart"
-                      android:singleLine="true"
-                      android:inputType="number" />
+                      android:hint="@string/bypass_hint"
+                      android:singleLine="true"/>
             </LinearLayout>
 
         </LinearLayout>
diff --git a/Ethernet/res/values/strings.xml b/Ethernet/res/values/strings.xml
index 404c297..564d7e2 100644
--- a/Ethernet/res/values/strings.xml
+++ b/Ethernet/res/values/strings.xml
@@ -32,4 +32,5 @@
     <string name="eth_proxy_address">Proxy address</string>
     <string name="show_need_setting">You should setting the static IP and DNS address in this mode</string>
     <string name="advanced_need_setting">You should not set the Proxy address and Proxy port null</string>
+    <string name="bypass_hint">You should put valid addresses split by \",\"</string>
 </resources>
-- 
1.8.0

